[vtk] Adjust VTK_VERSION macros for simplicity

In our Bazel rules we scrape vtk/CMake/vtkVersion.cmake for what it
contains (VTK_MAJOR_VERSION, VTK_MINOR_VERSION, VTK_BUILD_VERSION),
but that doesn't give us values for VTK_VERSION_FULL (nominally
scraped from git) nor VTK_EPOCH_VERSION (some upstream CI voodoo).
We need to patch those substitutions using what we do have access to.

This change is Drake-specific, so we do not plan to upstream it.

--- Common/Core/vtkVersionFull.h.in
+++ Common/Core/vtkVersionFull.h.in
@@ -5,6 +5,6 @@
 
 /* This is in its own header to reduce build dependencies */
 
-#define VTK_VERSION_FULL "@VTK_VERSION_FULL@"
+#define VTK_VERSION_FULL VTK_VERSION "-drake"
 
 #endif


--- Common/Core/vtkVersionQuick.h.in
+++ Common/Core/vtkVersionQuick.h.in
@@ -7,7 +7,7 @@
 
 #define VTK_MAJOR_VERSION @VTK_MAJOR_VERSION@
 #define VTK_MINOR_VERSION @VTK_MINOR_VERSION@
-#define VTK_EPOCH_VERSION @VTK_EPOCH_VERSION@
+#define VTK_EPOCH_VERSION @VTK_BUILD_VERSION@
 
 #define VTK_VERSION_CHECK(major, minor, build)                                                     \
   (10000000000ULL * major + 100000000ULL * minor + build)
